File: project-management-service/src/main/java/com/project/project_management_service/model/Task.java
package com.project.project_management_service.model;

import com.project.project_management_service.model.enums.Profession;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "tasks")
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID taskId;
    private String taskName;
    private String taskDescription;
    private String taskStatus;
    private String taskPriority;
    private UUID projectId;
    private String projectName;
    private UUID userId;
    private String username;
    private String profession;
}

File: task-management-service/src/main/kotlin/com/project/task_management_service/model/Task.kt
package com.project.task_management_service.model

import com.project.task_management_service.model.enums.TaskPriority
import com.project.task_management_service.model.enums.TaskStatus
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import lombok.AllArgsConstructor
import lombok.Data
import lombok.NoArgsConstructor
import java.util.*

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "task")
data class Task(
    @Id
    val taskId: UUID,
    val taskName: String,
    val taskDescription: String,
    val taskStatus: TaskStatus,
    val taskPriority: TaskPriority,
    val projectId: UUID,
    val projectName: String,
    val userId: UUID,
    val userName: String,
    val profession: String
)
